src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.cs

```diff
@@ -0,0 +1,459 @@
+                                                                                      description: s_localizableDescription,
+                                                                                      isPortedFxCopRule: false,
+                                                                                      isDataflowRule: false);
+
+        internal static DiagnosticDescriptor ObsoleteRule = DiagnosticDescriptorHelper.Create(RuleId,
+                                                                                      s_localizableTitle,
+                                                                                      s_localizableAddedMessage,
+                                                                                      DiagnosticCategory.Interoperability,
+                                                                                      RuleLevel.BuildWarning,
+                                                                                      description: s_localizableObsoleteMessage,
+                                                                                      isPortedFxCopRule: false,
+                                                                                      isDataflowRule: false);
+        internal static DiagnosticDescriptor RemovedRule = DiagnosticDescriptorHelper.Create(RuleId,
+                                                                                      s_localizableTitle,
+                                                                                      s_localizableAddedMessage,
+                                                                                      DiagnosticCategory.Interoperability,
+                                                                                      RuleLevel.BuildWarning,
+                                                                                      description: s_localizableRemovedMessage,
+                                                                                      isPortedFxCopRule: false,
+                                                                                      isDataflowRule: false);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(MinimumOsRule, ObsoleteRule, RemovedRule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterCompilationStartAction(context =>
+            {
+                var typeName = WellKnownTypeNames.SystemRuntimeInteropServicesRuntimeInformation;
+
+                if (!context.Compilation.TryGetOrCreateTypeByMetadataName(typeName + "Helper", out var runtimeInformationType))
+                {
+                    runtimeInformationType = context.Compilation.GetOrCreateTypeByMetadataName(typeName);
+                }
```

Comment:
Should we also bail if `SystemRuntimeInteropServicesRuntimeInformation` doesn't exist?

Reply:
Yes, but for now i cannot use new APIs in the test, so I am using mock `typeName + "Helper"` which has those new methods, after my tests able to load new APIs then i will change this to load the real `runtimeInformationType` and will bailout. When i use the analyzer for testing real code like `runtime` it shouldn't depend on `typeName + "Helper"` type